### ISSUE TITLE
Add react-native to already existing application throws errors

### ISSUE THREAD

**Issue by 814k31** [facebook/react-native#29315]
Please provide all the information requested. Issues that do not follow this format are likely to stall.

## Description
Hey Guys, I'm trying to follow the [docs](https://reactnative.dev/docs/integration-with-existing-apps) about adding react-native to an already existing android application.

I am receiving the error(s):
```
2020-07-09 17:07:12.964 6562-6562/com.blake.testreactnative E/SoLoader: Error when loading lib: dlopen failed: library "libjsc.so" not found lib hash: 37b75c7fd33e8c94378f65171f48bada search path is /data/app/com.blake.testreactnative-QIGbyGF2mRQe_c1Zn8Rq4A==/lib/x86
```

and also:
```
2020-07-09 17:07:13.334 6562-6623/com.blake.testreactnative E/SoLoader: couldn't find DSO to load: libhermes.so result: 0
2020-07-09 17:07:13.335 6562-6623/com.blake.testreactnative E/AndroidRuntime: FATAL EXCEPTION: create_react_context
    Process: com.blake.testreactnative, PID: 6562
    java.lang.UnsatisfiedLinkError: couldn't find DSO to load: libhermes.so result: 0
        at com.facebook.soloader.SoLoader.doLoadLibraryBySoName(SoLoader.java:825)
        at com.facebook.soloader.SoLoader.loadLibraryBySoName(SoLoader.java:673)
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:611)
        at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:559)
        at com.facebook.hermes.reactexecutor.HermesExecutor.<clinit>(HermesExecutor.java:20)
        at com.facebook.hermes.reactexecutor.HermesExecutorFactory.create(HermesExecutorFactory.java:29)
        at com.facebook.react.ReactInstanceManager$5.run(ReactInstanceManager.java:1017)
        at java.lang.Thread.run(Thread.java:764)
```

## React Native version:
Run `react-native info` in your terminal and copy the results here.
System:
    OS: macOS 10.15.5
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 61.69 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.18.2 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.14.5 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.8.4 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.5, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK:
      API Levels: 22, 23, 26, 27, 28, 29
      Build Tools: 27.0.3, 28.0.2, 28.0.3, 29.0.2, 29.0.3, 30.0.0, 30.0.0
      System Images: android-27 | Google Play Intel x86 Atom, android-28 | Wear OS Intel x86 Atom
      Android NDK: 18.1.5063045
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
    Xcode: 11.5/11E608c - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_242-release - /usr/bin/javac
    Python: 2.7.16 - /usr/local/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: ^0.63.0 => 0.63.0 
  npmGlobalPackages:
    *react-native*: Not Found

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

![react-native-bug](https://user-images.githubusercontent.com/9160551/87023438-49b0b380-c20a-11ea-98ce-2871942ecf0f.gif)

## Expected Results
I would expect the react native view to open and hello world would appear on the screen

## Snack, code example, screenshot, or link to a repository:
https://github.com/814k31/testReactNative

## Documentation Difficulty
I found the documentation quite confusing as it mentions `ReactInstanceManager` for the first time like it's already been talked about. The code mentioned in the image below I have placed inside of `MyReactActivity.kt` because rest of the documentation mentions using `ReactInstanceManager` there.

### Screenshot
<img width="925" alt="Screen Shot 2020-07-09 at 5 12 36 pm" src="https://user-images.githubusercontent.com/9160551/87021234-88913a00-c207-11ea-9d5a-ebc81ef9ba43.png">

**Comment by 814k31** [comment]
Fixed by adding to jsc to the projects dependencies and applying react-natives build.gradle in the apps build.gradle
https://github.com/814k31/testReactNative/pull/1

Perhaps the docs should be updated for others

**Comment by ajitsing** [comment]
This fix worked for me as well.

**Comment by cortinico** [comment]
This and all the "couldn't find DSO to load" bugs have been fixed in React Native 0.71.

The issue on previous versions was a "too aggressive" clean-up logic in the `react.gradle` file. In 0.71 the clean-up logic has been rewritten to be more Android friendly and is now living inside the React Native Gradle Plugin.

I'm closing this, please update to 0.71 and open a new issue if the "couldn't find DSO to load" crash is still happening on 0.71+.
